pinot-spi/src/main/java/org/apache/pinot/spi/trace/DefaultRequestContext.java

```diff
@@ -562,6 +563,16 @@ public void setProcessingExceptions(List<String> processingExceptions) {
     _processingExceptions.addAll(processingExceptions);
   }
 
+  @Override
+  public Map<String, List<String>> getRequestHttpHeaders() {
```

Comment:
this have some problem --> for example i highly doubt we should keep the entire request header in (even for debugging purposes) . for example, we surely dont want to keep the auth token in here

this prompts me to wonder if we should actually explicitly keep the part of the header explicitly instead of having a generic map.

Reply:
That is a good point! 

One solution I can think of is to add a broker-level config where we can give list of headers to parse from http-headers and push to request-context. Let me know if this sounds good or if there is any better alternative.

Reply:
hmm. this is a good question. having everything in there is simple. but I suppose what we really want here is customizable header tracking (similar to distributed tracing, yes?); where different organizations might have different ways to keep these in check. 

can we create an issue to discuss the scope? sorry to drag this long but it's better to understand this and get concensus 

my understanding is:

1. there are some extra metadata info related to the request we wanted to track in the query response
2. these metadata info might be easy to identify and aggregate certain query patterns (such as auth group name; client type; etc)
3. there can be customizable header fields we want to encode but there are certain ones we want to left out
4. there should be a way, at service startup, to determine 
    - what fields are expected to be recorded in the RequestContext, and 
    - what are each fields expected data type (e.g. number/string/etc)
5. retrieval of these RequestContext fields should be generic (Similar to `Properties`)

thoughts?

Reply:
Yes this sums up everything pretty much! Let me start an issue to get more inputs from the community if others have any special requirement(s).

Reply:
@walterddr updated the PR with a config for `allowlisted-http-headers` to parse.